reactive/documentation/1.0/reference/html_single/index.adoc

```diff
@@ -0,0 +1,15 @@
+= Reference Documentation for 1.0
+:awestruct-layout: project-standard
+:awestruct-project: reactive
+:page-interpolate: true
+:project: #{site.projects[page.project]}
+:latest_release: #{latest_release(page).version}
+:toc:
+:numbered:
+:leveloffset: +1
+
+include::https://raw.githubusercontent.com/hibernate/hibernate-reactive/{latest_release}/documentation/src/main/asciidoc/reference/preface.adoc[]
```

Comment:
Couldn't we rather have a dedicated repo where we push our documentation as part of the project builds (reactive/...)? 

Then you could just copy the whole repo to `./_site` as part of the `hibernate.org` build.

Reply:
I like the simplicity of this solution but it's not possible to apply it with the javadoc.
So I will try that approach next.
But I'm not sure what you mean by dedicated repository. Do you mean publish the doc somewhere? Similar to what we do now with JBoss.org?

Reply:
I was thinking to create a dedicated repository (say, `hibernate:docs.hibernate.org`) to host all rendered documentation files. We would push to that repository during the project builds (Search, Reactive, ...), instead of rsync'ing to JBoss.org.

Then when building https://github.com/hibernate/hibernate.org/, we would clone `hibernate:docs.hibernate.org` to a temporary directory, copy the relevant sub-directories to `./_publish_tmp/`, and all the documentation would automatically be published along with the website when we rsync.

In practice, this means we would add a line before [this one](https://github.com/hibernate/hibernate.org/blob/staging/_scripts/publish-to-production.sh#L38) to copy stuff from `hibernate:docs.hibernate.org` to `./_publish_tmp/`

Alternatively, we could have project builds publish the documentation directly to repository https://github.com/hibernate/hibernate.org/, say in a `_documentation` directory. I'm not sure if there is a downside?

Other solution, we could do what Quarkus does: essentially, when releasing quarkus, they push the asciidoc files (not the rendered files) to the source repository of their website (https://github.com/quarkusio/quarkusio.github.io/), like this: https://github.com/quarkusio/quarkusio.github.io/commit/19c78a3f8862f8192fdc6ed8e5e8ca902bee3bc5

The main problem with this solution is that you need to do the rendering of the documentation when building the website. In the case of Quarkus, they only keep documentation for the latest version, so that's fine. But in our case it means the amount of documents to render will grow a lot as we add more versions; I'm not sure it will scale that well. Also, changes in asciidoc versions might be a problem in the long run (old versions of the doc might not render exactly the same).

If it was just me, I would be in favor of pushing pre-rendered (HTML) files to `hibernate/hibernate.org`. Maybe @Sanne has an opinion on this?

Reply:
One thing to decide is how we want the documentation to look like:

1. Do we want to keep the current look: https://docs.jboss.org/hibernate/orm/5.4/userguide/html_single/Hibernate_User_Guide.html
2. or we want it rendered like it's part of our website:  https://staging.hibernate.org/reactive/documentation/1.0/reference/html_single/
Note that this solution require us to tweak the CSS because currently it doesn't look terribly good 


> If it was just me, I would be in favor of pushing pre-rendered (HTML) files to hibernate/hibernate.org.
I will try this approach.

> Alternatively, we could have project builds publish the documentation directly to repository https://github.com/hibernate/hibernate.org/, say in a _documentation directory. I'm not sure if there is a downside?

Reply:
> if it was just me, I would be in favor of pushing pre-rendered (HTML) files to hibernate/hibernate.org

I think I prefer this one as well

Reply:
> I was trying to make the documentation appear like it's part of the website. But the reality is that this would require a lot of work on the CSS.

Yeah I noticed that the other time when trying to push a complex migration guide to hibernate.org. I just gave up and moved the migration guide to the Hibernate Search repository. Working on the CSS doesn't seem worth the amount of work, and I'm sure it'll be an endless source of trouble.